### PR TITLE
Preview Cleanup + Seperating Components

### PR DESCRIPTION
**Pull request by Rahkeen**
This PR seperates out feature components from the screen file. It's actually a bit cumbersome to leverage previews when there are a BUNCH of previews in the same file. Also discoverability suffers, so I moved a bunch of things out into their own component files. This also makes it easy to see the various states of each component.

It also isolates the screen files to only have previews at the screen level.

### CODE REVIEW

**Comment by Rahkeen** [comment]
* **#120** <a href="https://app.graphite.dev/github/pr/EmergeTools/hackernews/120?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Rahkeen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rbro112 — APPROVED** [review]
Awesome stuff!

File: android/app/src/main/java/com/emergetools/hackernews/ui/preview/AppStoreSnapshot.kt [hunk 1]
```diff
@@ -3,9 +3,28 @@ package com.emergetools.hackernews.ui.preview
...
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewLightDark
 import com.emergetools.snapshots.annotations.EmergeAppStoreSnapshot
 
 @EmergeAppStoreSnapshot
 @Preview(device = Devices.PIXEL_5, uiMode = UI_MODE_NIGHT_YES)
 @Preview(device = Devices.PIXEL_5)
 annotation class AppStoreSnapshot
+
+@Preview(
```
Comment (rbro112) [inline android/app/src/main/java/com/emergetools/hackernews/ui/preview/AppStoreSnapshot.kt:14]: Seems we're having some trouble with some of these. Any chance you could use a default device - maybe skip the landscape temporarily and do:
`device = Devices.PIXEL_FOLD`
and
`device = Devices.PIXEL_TABLET`

Comment (Rahkeen) [reply android/app/src/main/java/com/emergetools/hackernews/ui/preview/AppStoreSnapshot.kt:14]: Done

**Pull request merged**